From: Jeff Mahoney <jeffm@suse.com>
Subject: xfs: fixed signedness of error code in xfs_inode_buf_verify
References: bsc#1003153
Patch-mainline: Submitted to stable@vger.kernel.org, bug in backport

Prior to v3.17, XFS used positive error codes internally.  Commits
backported from later releases need the error code signs reversed.
Negative error codes will cause assertion failures (BUG_ON on kernels
with assertions enabled.

Cc: <stable@vger.kernel.org> # v3.12-v3.16
Signed-off-by: Jeff Mahoney <jeffm@suse.com>
---
 fs/xfs/xfs_inode_buf.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

--- a/fs/xfs/xfs_inode_buf.c
+++ b/fs/xfs/xfs_inode_buf.c
@@ -101,7 +101,7 @@ xfs_inode_buf_verify(
 						XFS_RANDOM_ITOBP_INOTOBP))) {
 			if (readahead) {
 				bp->b_flags &= ~XBF_DONE;
-				xfs_buf_ioerror(bp, -EIO);
+				xfs_buf_ioerror(bp, EIO);
 				return;
 			}
 
